js/sdks/packages/cli/src/utils/exec.ts

```diff
@@ -32,9 +48,76 @@ export function runCommand(command: string, streamStdio?: boolean): RunCommandRe
     const r = execSync(`${command}`, options);
     return {
       status: 0,
-      stdout: r,
+      stdout: r.toString(),
     };
   } catch (e) {
-    return e as RunCommandResult;
+    const err = e as RunCommandError;
+    return {
+      status: err.status,
+      message: err.message,
+      stderr: err.stderr ? err.stderr.toString() : '',
+      stdout: err.stdout ? err.stdout.toString() : '',
+    };
+  }
+}
+
+function timeout(delay: number) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, delay);
+  });
+}
+
+export function runCommandWithHealthcheck(command: string, options: RunCommandWithHealthcheckOptions) {
+  const cmd = spawn('sh', ['-c', command]);
+
+  cmd.stdout.on('data', (data: Buffer) => {
+    const strData = data.toString();
+    if (options.streamStdout) {
+      console.log(strData);
+    }
+
+    if (options.onStdin) {
+      options.onStdin.next(strData);
+    }
+  });
+
+  cmd.stderr.on('data', (data: string) => {
+    console.error(data.toString());
+  });
+
+  cmd.on('close', (code) => {
+    if (!options.exitProcess) {
+      return;
+    }
+
+    if (code !== null) {
+      process.exit(code);
+    }
+    process.exit(0);
+  });
+
+  process.on('SIGINT', function () {
```

Comment:
should we just go ahead and capture every possible exit event for cleanup purposes?

Reply:
Also I'm not sure where cleanup happens but we did have that issue before where it looks like it hangs even though its done cleaning up, maybe need to flush somehow?